pallets/relayer/src/lib.rs

```diff
@@ -72,84 +71,100 @@ pub trait WeightInfo {
+        UpdatedPolyxLimit(EventDid, T::AccountId, T::AccountId, Balance, Balance),
     }
-}
 
-/// A Subsidy for transaction and protocol fees.
-///
-/// This holds the subsidiser's paying key and the remaining POLYX balance
-/// available for subsidising transaction and protocol fees.
-#[derive(Encode, Decode, TypeInfo)]
-#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Subsidy<Acc> {
-    /// The subsidiser's paying key.
-    pub paying_key: Acc,
-    /// How much POLYX is remaining for subsidising transaction and protocol fees.
-    pub remaining: Balance,
-}
-
-/// Update action for subsidy POLYX limit.
-enum UpdateAction {
-    /// Set the current subsidy limit to `amount`.
-    Set,
-    /// Add `amount` to the current subsidy limit.
-    Add,
-    /// Subtract `amount` from the current subsidy limit.
-    Sub,
-}
+    /// A Subsidy for transaction and protocol fees.
+    ///
+    /// This holds the subsidiser's paying key and the remaining POLYX balance
+    /// available for subsidising transaction and protocol fees.
+    #[derive(
+        Encode,
+        Decode,
+        TypeInfo,
+        Clone,
+        Debug,
+        Default,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        MaxEncodedLen
+    )]
+    pub struct Subsidy<Acc> {
+        /// The subsidiser's paying key.
+        pub paying_key: Acc,
+        /// How much POLYX is remaining for subsidising transaction and protocol fees.
+        pub remaining: Balance,
+    }
 
-decl_storage! {
-    trait Store for Module<T: Config> as Relayer {
-        /// The subsidy for a `user_key` if they are being subsidised,
-        /// as a map `user_key` => `Subsidy`.
-        ///
-        /// A key can only have one subsidy at a time.  To change subsidisers
-        /// a key needs to call `remove_paying_key` to remove the current subsidy,
-        /// before they can accept a new subsidiser.
-        pub Subsidies get(fn subsidies):
-            map hasher(blake2_128_concat) T::AccountId => Option<Subsidy<T::AccountId>>;
+    /// Update action for subsidy POLYX limit.
+    #[derive(Copy, Clone)]
+    pub enum UpdateAction {
+        /// Set the current subsidy limit to `amount`.
+        Set,
+        /// Add `amount` to the current subsidy limit.
+        Add,
+        /// Subtract `amount` from the current subsidy limit.
+        Sub,
     }
-}
 
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::RuntimeOrigin {
-        type Error = Error<T>;
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
 
-        fn deposit_event() = default;
+    /// The subsidy for a `user_key` if they are being subsidised,
+    /// as a map `user_key` => `Subsidy`.
+    ///
+    /// A key can only have one subsidy at a time.  To change subsidisers
+    /// a key needs to call `remove_paying_key` to remove the current subsidy,
+    /// before they can accept a new subsidiser.
+    #[pallet::storage]
```

Comment:
This storage is missing the getter

Reply:
Substrate is going to deprecate the getter attribute: https://github.com/paritytech/polkadot-sdk/issues/3326
They have started to remove use of it from the Substrate pallets.  We should start doing the same.  For the v2 upgrade I didn't want to include that change.  This will mostly impact our unit tests and benchmark code.